firewood/src/db.rs

```diff
@@ -79,7 +77,7 @@ impl<T: ReadLinearStore> api::DbView for HistoricalRev<T> {
     async fn single_key_proof<K: api::KeyType>(
         &self,
         _key: K,
-    ) -> Result<Option<Proof<Vec<u8>>>, api::Error> {
+    ) -> Result<Option<Proof>, api::Error> {
```

Comment:
Removing this generic means each ProofNode can be of a different type, which isn't what we want I think. All ProofNodes within a proof should either be owned or borrowed (currently all of them are borrowed).

I'd rather see the generic here and have ProofNode use the type from it's parent.

Reply:
The generic is added back in https://github.com/ava-labs/firewood/pull/682/files. I could merge that PR into this one if you want.